rust/src/rfb/rfb.rs

```diff
@@ -549,10 +628,8 @@ impl RFBState {
                         Ok((_rem, request)) => {
                             if let Some(current_transaction) = self.get_current_tx() {
                                 current_transaction.tc_failure_reason = Some(request);
-                            } else {
-                                return AppLayerResult::err();
                             }
-                            return AppLayerResult::err();
```

Comment:
oh @jasonish is there a linter to catch this ?
```
} else {
return x;
}
return x;
```

Reply:
> oh @jasonish is there a linter to catch this ?
> 
> ```
> } else {
> return x;
> }
> return x;
> ```

If the second one is dead code, rust will warn about it by default.

Reply:
No dead code but it can be simplified to `return x` ;-)

Reply:
I don't see a lint here that looks like it would do this: https://rust-lang.github.io/rust-clippy/master/index.html#/return